casper/src/main/scala/coop/rchain/casper/api/BlockApiImpl.scala

```diff
@@ -175,6 +181,60 @@ class BlockApiImpl[F[_]: Concurrent: RuntimeManager: BlockDagStorage: BlockStore
       )
   }
 
+  override def deployStatus(deployId: DeployId): F[ApiErr[DeployExecStatus]] = {
+    def withSeqParAndBlockInfo(
+        blockHashOpt: Option[BlockHash]
+    )(f: (Seq[Par], BlockInfo) => DeployExecStatus): F[DeployExecStatus] = {
+      import coop.rchain.models.rholang.implicits._
+      for {
+        blockHash <- blockHashOpt.map(_.toHexString).get.pure
+        par       <- Sync[F].delay(GDeployId(deployId))
+        dataAtPar <- getDataAtPar(par, blockHash, usePreStateHash = false)
+        seqPar    <- liftToBlockApiErr(dataAtPar).map { case (seqPar, _) => seqPar }
+        blockInfo <- getBlock(blockHash) >>= liftToBlockApiErr
+      } yield f(seqPar, blockInfo)
+    }
+
+    def notProcessed(status: String): F[DeployExecStatus] =
+      DeployExecStatus().withNotProcessed(NotProcessed(status)).pure
+
+    def unknown = "Unknown"
+
+    def statusF: F[DeployExecStatus] =
+      for {
+        blockHashOpt    <- BlockDagStorage[F].lookupByDeployId(deployId)
+        isDeployInBlock = blockHashOpt.isDefined
+        isDeployInPool  <- BlockDagStorage[F].pooledDeploys.map(_.contains(deployId))
```

Comment:
If the deployment is not found in the DAG (not included in the block), it may be in processing (pooled). This flag is needed to inform the user of the appropriate status.